### PR TITLE
Tidy up more 0.11.0 docs

### PR DESCRIPTION
**Pull request by lauralorenz**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [n/a] adds new tests (if appropriate)
- [n/a] updates `CHANGELOG.md` (if appropriate)
- [n/a] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Update the docs for more 0.11.0 goodness including
- [x] API docs header for results and result
- [x] Deprecation notes on docs header for Result Handlers
- [x] Updates/comb through of Secrests docs including Secrets API docs

## Why is this PR important?
Unify our messaging our secrets and results as of 0.11.0.

### CODE REVIEW

**Comment by joshmeek** [comment]
@lauralorenz For the secrets stuff there's a lot in the new authentication recipe @cicdw wrote https://docs.prefect.io/orchestration/recipes/third_party_auth.html that could be used elsewhere

File: src/prefect/engine/result_handlers/__init__.py [hunk 1]
```diff
@@ -1,4 +1,8 @@
 """
+:::warning
+Result handlers have been deprecated in 0.11.0. For Prefect installations 0.11.00+, at runtime, all result handlers configured on a flow will be autoconverted into a matching subclass of `Result` from `prefect.engine.results`.
```
Comment (cicdw) [inline src/prefect/engine/result_handlers/__init__.py:3]:
```suggestion
Result handlers have been deprecated in 0.11.0. For Prefect installations 0.11.0+, at runtime, all result handlers configured on a flow will be autoconverted into a matching subclass of `Result` from `prefect.engine.results`.
```

File: src/prefect/engine/results/__init__.py [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+
+@task
+def my_example_task():
+    # read data from a file in the bucket
+    my_task_data = MY_RESULTS.read(location="some_data_in_my_bucket.csv")
+    print(my_task_data.value) # is the deserialized data that was in the file s3://my_bucket/some_data_in_my_bucket.csv
+    
+    # write data to a templated filename in the bucket
+    data = 3
+    my_task_data = MY_RESULTS.write(data, location="{task_name}.txt")
```
Comment (cicdw) [inline src/prefect/engine/results/__init__.py:24]: I think you need to explicitly provide `**kwargs` to `write` in order for any templating to take place

File: src/prefect/engine/results/__init__.py [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+    print(my_task_data.value) # is the value of the variable data
+    print(my_task_data.location) # is "my_example_task.txt"
+
+```
+
+Results will only persist data if checkpointing is turned on. To learn more about how to use results and how to configure checkpointing, read our tutorial on [Using Results](../core/advanced_tutorials/using-results.md).
```
Comment (cicdw) [inline src/prefect/engine/results/__init__.py:30]:
```suggestion
Results will only persist return data if checkpointing is turned on. To learn more about how to use results and how to configure checkpointing, read our tutorial on [Using Results](../core/advanced_tutorials/using-results.md).
```

File: src/prefect/engine/results/__init__.py (hunk 2 above)
Comment (lauralorenz) [reply src/prefect/engine/results/__init__.py]: Fixed in b26d2c1799347e2a4bf9efddaca0bf04f0fceac5

**Comment by lauralorenz** [comment]
Update re: secrets: mostly was just moving secrets stuff around more than anything else (especially the default secrets are in the Secrets concepts documentation first and foremost and linked to in the deployment recipe), but referenced the new third-party auth docs in several other places and added some examples.

File: docs/core/concepts/secrets.md [hunk 4]
```diff
@@ -17,15 +19,32 @@ Though Prefect takes steps to ensure that `Secret` objects do not reveal sensiti
...
+For example, given an environment with the environment variable `PREFECT__CONTEXT__SECRETS__FOO=mypassword`, the value `"mypassword"` could be retrieved by using a `PrefectSecret` task or by using the secrets API directly, as shown below.
 
-If the secret is not found in local context and `config.cloud.use_local_secrets=False`, the base `Secret` class queries the Prefect Cloud API for a stored secret. This call can only be made successfully by authenticated Prefect Cloud Agents.
+:::: tabs
+::: tab PrefectSecret
+```python
+>>> from prefect.tasks.secrets import PrefectSecret
+>>> p = PrefectSecret('foo')                        
+>>> p.run()                   
+{'ACCESS_KEY': 'abcdef', 'SECRET_ACCESS_KEY': 'ghijklmn'}
```
Comment (cicdw) [inline docs/core/concepts/secrets.md:32]: Might want to make this output the same value as the other tab

File: docs/orchestration/recipes/third_party_auth.md [hunk 5]
```diff
@@ -78,9 +78,8 @@ This will then ensure that the `AWS_CREDENTIALS` secret is globally present in `
...
 
-### List of default Secret names
+### Default Secret names
 
-The following is a list of the default names and contents of Prefect Secrets that, if set and declared, can be used to automatically authenticate your flow with the listed service:
+A few common secrets, such as authentication keys for GCP or AWS, have a standard naming convention as Prefect secrets for use by the Prefect pipeline or tasks in Prefect's task library. If you follow this naming convention when storing your secrets, all supported Prefect interactions with those services will be automatically configured.
 
-- `GCP_CREDENTIALS`: a dictionary containing a valid [Service Account Key](https://cloud.google.com/docs/authentication/getting-started)
-- `AWS_CREDENTIALS`: a dictionary containing two required keys: `ACCESS_KEY` and `SECRET_ACCESS_KEY`, and an optional `SESSION_TOKEN`, which are passed directly to [the `boto3` client](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html)
+See the list of defualt secret names on the [Secrets concept documentation](../../core/concepts/secrets.md#default-secrets).
```
Comment (cicdw) [inline docs/orchestration/recipes/third_party_auth.md:85]:
```suggestion
See the list of default secret names on the [Secrets concept documentation](../../core/concepts/secrets.md#default-secrets).
```

File: src/prefect/client/secrets.py [hunk 6]
```diff
@@ -41,6 +41,10 @@
 export PREFECT__CONTEXT__SECRETS__MY_KEY="MY_VALUE"
 ```
 
+::: tip Default secrets
+Special default secret names can be used to authenticate to third-party systems in a installation-wide way. Read more about this in our [Secrets concept documentation](../../../core/secrets.md#default-secrets).
```
Comment (cicdw) [inline src/prefect/client/secrets.py:45]: FYI I recently learned that if you provide a doc path staring with `/` from the root you can avoid getting the directory structure right.  So in this case, `"/core/secrets.md#default-secrets"` should do the trick

Comment (lauralorenz) [reply src/prefect/client/secrets.py]: Ahhh this is so much better (used here in cc15fb5)

File: docs/core/concepts/secrets.md (hunk 4 above)
Comment (lauralorenz) [reply docs/core/concepts/secrets.md]: Fixed in cc15fb5, I had too much AWS on the brain

**Pull request merged**